From 359614b22318cbc2e9773a62ed94ba26a4da2dd8 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Mon, 28 Sep 2026 03:54:05 +0100
Subject: [PATCH] drop the bootclasspath fragment from the virt apex

the com.android.virt apex is no longer installed on the gsi. soong
still analyses the apex variant of the bootclasspath fragment and
rejects the build unless framework-virtualization is declared in
PRODUCT_APEX_BOOT_JARS, which is pointless for an apex that is never
mounted. remove the fragment and its apex reference; the
framework-virtualization java_sdk_library stays so the framework and
Apps that compile against its stubs keep building, with the classes
served statically via framework-non-updatable-unbundled-impl-libs.
---
 build/apex/Android.bp | 43 -------------------------------------------
 1 file changed, 43 deletions(-)

diff --git a/build/apex/Android.bp b/build/apex/Android.bp
index f24eb80..412e8a2 100644
--- a/build/apex/Android.bp
+++ b/build/apex/Android.bp
@@ -49,9 +49,6 @@ apex_defaults {
 
     file_contexts: ":com.android.virt-file_contexts",
 
-    bootclasspath_fragments: [
-        "com.android.virt-bootclasspath-fragment",
-    ],
     jni_libs: [
         "libvirtualizationservice_jni",
         "libvirtualmachine_jni",
@@ -342,46 +339,6 @@ python_binary_host {
     ],
 }
 
-// Encapsulate the contributions made by the com.android.virt to the bootclasspath.
-bootclasspath_fragment {
-    name: "com.android.virt-bootclasspath-fragment",
-    contents: ["framework-virtualization"],
-    apex_available: ["com.android.virt"],
-
-    // The bootclasspath_fragments that provide APIs on which this depends.
-    fragments: [
-        {
-            apex: "com.android.art",
-            module: "art-bootclasspath-fragment",
-        },
-    ],
-
-    // Additional stubs libraries that this fragment's contents use which are
-    // not provided by another bootclasspath_fragment.
-    additional_stubs: [
-        "android-non-updatable",
-    ],
-
-    hidden_api: {
-
-        // This module does not contain any split packages.
-        split_packages: [],
-
-        // The following packages and all their subpackages currently only
-        // contain classes from this bootclasspath_fragment. Listing a package
-        // here won't prevent other bootclasspath modules from adding classes in
-        // any of those packages but it will prevent them from adding those
-        // classes into an API surface, e.g. public, system, etc.. Doing so will
-        // result in a build failure due to inconsistent flags.
-        package_prefixes: [
-            "android.system.virtualmachine",
-            "android.system.virtualizationservice",
-            // android.sysprop.*, renamed by jarjar
-            "com.android.system.virtualmachine.sysprop",
-        ],
-    },
-}
-
 systemserverclasspath_fragment {
     name: "com.android.virt-systemserver-fragment",
     contents: [
-- 
2.55.0

